docs/tutorials/morpheusvm/intro.md

```diff
@@ -0,0 +1,766 @@
+
+To get started, clone the HyperSDK:
+
+```sh
+git clone git@github.com:ava-labs/hypersdk.git
+cd hypersdk
+```
+
+## Initializing MorpheusVM
+
+To start, we'll create a new directory in examples and initialize the go mod:
+
+```sh
+mkdir examples/tutorial
+cd examples/tutorial
+go mod init
+```
+
+Now that we've , we'll add a `consts.go` file to declare the name, HRP for Bech32 addresses,
+and the initial version of our VM:
+
+```golang
+package tutorial
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/version"
+)
+
+const (
+	HRP  = "tutorial"
+	Name = "tutorialvm"
+)
+
+var ID ids.ID
+
+func init() {
+	b := make([]byte, ids.IDLen)
+	copy(b, []byte(Name))
+	vmID, err := ids.ToID(b)
+	if err != nil {
+		panic(err)
+	}
+	ID = vmID
+}
+
+var Version = &version.Semantic{
+	Major: 0,
+	Minor: 0,
+	Patch: 1,
+}
+
+```
+
+This will create a warning because we have not imported AvalancheGo into this workspace yet.
+To fix that, let's add AvalancheGo as a dependency:
+
+```sh
+go get github.com/ava-labs/avalanchego && go mod tidy
+```
+
+## Creating `Transfer` Pt. 1
+
+We'll implement a single `Transfer` action to handle payments in a native token.
+Actions are wrapped into HyperSDK transactions and executed on-chain, so the struct
+itself will include all of the fields needed to verify and execute onchain.
+
+You can think of each field in the `Transfer` type as if it's a parameter to the function
+you want to execute onchain.
+
+To start, your `transfer.go` file will declare the struct and add a type assertion that it
+implements the `chain.Action` interface:
+
+
+```golang
+package tutorial
+
+import (
+   "github.com/ava-labs/hypersdk/chain"
+   "github.com/ava-labs/hypersdk/codec"
+)
+
+var _ chain.Action = (*Transfer)(nil)
+
+type Transfer struct {
+	// To is the recipient of the [Value].
+	To codec.Address `serialize:"true" json:"to"`
+
+	// Amount are transferred to [To].
+	Value uint64 `serialize:"true" json:"value"`
+
+	// Optional message to accompany transaction.
+	Memo []byte `serialize:"true" json:"memo"`
+}
+```
+
+This imports the HyperSDK, so we'll import it and redirect it to our local version of the
+HyperSDK:
+
+```sh
+go get github.com/ava-labs/hypersdk
+go mod edit -replace github.com/ava-labs/hypersdk=../../
+go mod tidy
+```
+
+Note: we include JSON and `serialize` tags in the struct to provide instructions on how
+the `Transfer` action should be marshalled to JSON or a binary representation. The JSON tags
+provide the key in the resulting JSON and `serialize:"true"` indicates that the codec should
+include each field when it marshals the struct to its binary representation.
+
+Right now, the `chain.Action` type assertion should fail because we haven't implemented the
+interface yet. To fix that for now, we'll include stubs for each of the required functions that
+call `panic("unimplemented")`, so we can come back to fill these in later in the tutorial.
+
+You can copy-paste the code below:
```

Comment:
VS code gets angry when we do this; although reloading it will make it
happy again